File: SWIFT/Cap07/c07ex06.swift
// c07ex06.swift

import Foundation

func Fatorial(_ N: Int) -> Int {
    var Fat = 1
    for I in 1...N {
        Fat *= I
    }
    return Fat
}

var Limite: Int = 0

repeat {
    print("Qual fatorial: ", terminator: "")
    if let entraLimite = Int(readLine() ?? "") {
        Limite = entraLimite
        print("Fatorial = \(Fatorial(Limite))")
        break
    }
    else {
        print("\nEntrada invalida. Tente novamente.\n")
    }
} while (true)

print()
print("Tecle <Enter> para encerrar...", terminator: "")
_ = readLine()

File: SCALA/Cap07/c07ex06.scala
// c07ex06.scala

import scala.io.StdIn._

object Main {
  def fatorial(N: Int): Int = {
    var fat = 1
    for (i <- 1 to N) {
      fat *= i
    }
    return fat
  }

  def main(args: Array[String]): Unit = {
    var limite: Int = 0  

    print("Qual fatorial: ")
    limite = readInt()

    println(s"Fatorial = ${fatorial(limite)}")

    print("\nTecle <Enter> para encerrar... ")
    readLine()
  }
}
